src/pages/NewsList/NewsList.tsx

```diff
@@ -104,8 +101,30 @@ export default function NewsList() {
 					))}
 				</div>
 			) : (
```

Comment:
I would move this condition before the return of this function. You can do something like:
```
const data = newsData && newsData.length > 0 ? newsData : JSON.parse(cachedData); 
```
(or a more complex condition, but in the end, you will end up with one variable with the data) 

and then just loop over the new variable (or check if it is empty)



or if you need to keep the conditions like they are now, you can create a new component something like NewsListItem which you would pass all the needed data.

The goal is to not repeat the HTML of the news item.